src/transaction-runner.coffee

```diff
@@ -66,8 +66,22 @@ class TransactionRunner
 
         catch error
           if error instanceof chai.AssertionError
-            data.message = "Failed assertion in hooks: " + error.message
-            data.test?.status = 'fail'
+            message = "Failed assertion in hooks: " + error.message
+
+            data['results'] ?= {}
+            data['results']['general'] ?= []
+            data['results']['general'].push { severity: 'error', message: message }
```

Comment:
It's a general error for the test step. It's not related to `body`, `headers`  or `statusCode`. Where do you propose to place it?